File: <100/020/020.hs
main :: IO ()
main = print $ solve start


solve :: Integer -> Integer
solve = digitSum . factorial


factorial:: Integer -> Integer
factorial 1 = 1
factorial n = n * factorial (n-1)


digitSum:: Integer -> Integer
digitSum 0 = 0
digitSum n = a + digitSum n'
    where
        a  = n `mod` 10
        n' = n `div` 10

start :: Integer
start:: Integer = 100

File: <100/020/020.jl
function digitSum(n:: BigInt):: Int
  if n == 0
    return 0
  else
    return n%10 + digitSum(n÷10)
  end
end


function main():: Nothing
  n:: BigInt = big(100)
  n |> factorial |> digitSum |> println

  return nothing
end


@time main()
